tests/framework/cli/test_starters.py

```diff
@@ -906,7 +895,29 @@ def test_directory_flag_with_starter_alias(self, fake_kedro_cli):
 class TestAddOnsFromUserPrompts:
     @pytest.mark.parametrize(
         "add_ons",
-        ["1", "2", "3", "4", "5", "6", "none", "2,3,4", "3-5", "all"],
+        [
+            "1",
+            "2",
+            "3",
+            "4",
+            "5",
+            "6",
+            "none",
+            "2,3,4",
+            "3-5",
+            "all",
+            "1, 2, 3",
+            "1,    2,   3",
+            "  1,  2, 3  ",
+            " 1-3 ",
+            "  1 - 3  ",
+            "ALL",
+            "All",
+            "AlL",
+            "NONE",
+            "None",
+            "nonE",
+        ],
```

Comment:
Some of these tests are repetitive, can it be reduce some of these to one example. For example
```
            "ALL",
            "All",
            "AlL" 
 ```
can be reduced to one since `AlL` is probably the most weird case?

Reply:
I do agree `AlL` is probably the strangest case, though I'd avoid simplifying to just either `ALL` or `All` as I'd want to ensure all small letters, all capitals, and a mix of the two are all processed correctly. `AlL` and `nonE` have both been dropped in [5ad8bab](https://github.com/kedro-org/kedro/pull/3263/commits/5ad8babe5a2f23eba1fe9ef30866f9cd6af254ff)